### PR TITLE
Postprocessing refactor all others

### PR DESCRIPTION
**Pull request by yiyixuxu**
wip. still need to fix some tests 

https://github.com/huggingface/diffusers/pull/2943

### CODE REVIEW

**Review by patrickvonplaten — COMMENTED** [review]
Looks very nice so far - @yiyixuxu let me know once you want a final review, once everything is ready :-) 

We could also think a bit about how we can advertise this bigger change in the docs

**Comment by yiyixuxu** [comment]
@patrickvonplaten ready now

File: src/diffusers/pipelines/deepfloyd_if/pipeline_if_img2img_superresolution.py [hunk 1]
```diff
@@ -762,7 +762,7 @@ def preprocess_image(self, image: PIL.Image.Image, num_images_per_prompt, device
             image = [np.array(i).astype(np.float32) / 255.0 for i in image]
 
             image = np.stack(image, axis=0)  # to np
-            torch.from_numpy(image.transpose(0, 3, 1, 2))
+            image = torch.from_numpy(image.transpose(0, 3, 1, 2))
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/deepfloyd_if/pipeline_if_img2img_superresolution.py:765]: Thanks for fixing!

**Comment by patrickvonplaten** [comment]
Looks great! Extensive tests are added, everything PRs is clean. Merging now to not run into merge conflicts right away again.

**Pull request merged**